tests/end2end/utils.py

```diff
@@ -0,0 +1,221 @@
+    """
+    Return the session related command line options for zhmc from an HMC
+    definition.
+
+    Parameters:
+
+      hmc_definition(zhmcclient.testutils.HMCDefiniton): HMC definition.
+
+    Returns:
+      list of str: zhmc session related command line options.
+    """
+    ret = []
+    if not hmc_definition.verify:
+        ret.append('-n')
+    ret.extend(['-h', hmc_definition.host])
+    ret.extend(['-u', hmc_definition.userid])
+    ret.extend(['-p', hmc_definition.password])
+    if hmc_definition.ca_certs:
+        ret.extend(['-c', hmc_definition.ca_certs])
+    return ret
+
+
+def run_zhmc(args, env=None, pdb=False):
+    """
+    Run the zhmc command and return its exit code, stdout and stderr.
+
+    Parameters:
+
+      args(list/tuple of str): List of command line arguments, not including
+        the 'zhmc' command name itself.
+
+      env(dict of str/str): Environment variables to be used instead of the
+        current process' variables.
+
+      pdb(bool): If True, debug the zhmc command. This is done by inserting
+        the '--pdb' option to the command arguments, and by not capturing the
+        stdout/stderr.
+
+    Returns:
+      tuple (rc, stdout, stderr) as follows:
+      - rc(int): zhmc exit code
+      - stdout(str): stdout string, or None if debugging the zhmc command
+      - stderr(str): stderr string, or None if debugging the zhmc command
+    """
+    assert isinstance(args, (list, tuple))
+    if env is not None:
+        assert isinstance(env, dict)
+
+    p_args = ['zhmc'] + args
+
+    # Set up output capturing, dependent on pdb flag
+    if pdb:
+        kwargs = {}
+        p_args.insert(1, '--pdb')
+    else:
+        kwargs = {
+            'stdout': subprocess.PIPE,
+            'stderr': subprocess.PIPE,
+        }
+
+    # pylint: disable=consider-using-with
+    proc = subprocess.Popen(p_args, env=env, **kwargs)
+
+    stdout, stderr = proc.communicate()
+    rc = proc.returncode
+    if six.PY3 and not pdb:
+        stdout = stdout.decode()
+        stderr = stderr.decode()
+
+    return rc, stdout, stderr
+
+
+def _res_name(item):
+    """Return the resource name, used by pick_test_resources()"""
+    if isinstance(item, (tuple, list)):
+        return item[0].name
+    return item.name
+
+
+def pick_test_resources(res_list):
+    """
+    Return the list of resources to be tested.
+
+    The env.var "TESTRESOURCES" controls which resources are picked for the
+    test, as follows:
+
+    * 'random': (default) one random choice from the input list of resources.
+    * 'all': the complete input list of resources.
+    * '<pattern>': The resources with names matching the regexp pattern.
+
+    Parameters:
+      res_list (list of zhmcclient.BaseResource or tuple thereof):
+        List of resources to pick from. Tuple items are a resource and its
+        parent resources.
+
+    Returns:
+      list of zhmcclient.BaseResource: Picked list of resources.
+    """
+
```

Comment:
minor: inconsistent ... normally you dont have an empty line after """

Reply:
I try to have an empty line in larger functions/methods that have further empty lines, and no empty line if the rest of the function/method is short enough to also not have further empty lines.

This case follows that rule, but I agree that it is not always done consistently.